classes/Kohana/Jam/Association/Hasmany.php

```diff
@@ -213,36 +229,34 @@ public function nullify_query(Jam_Model $model)
 	 */
 	public function remove_items_query(Jam_Model $model, array $ids)
 	{
-		switch ($this->delete_on_remove)
+		if (TRUE === $this->delete_on_remove OR Jam_Association::DELETE === $this->delete_on_remove)
 		{
-			case TRUE:
-			case Jam_Association::DELETE:
-				foreach (Jam::all($this->foreign_model)->where_key($ids) as $item )
-				{
-					$item->delete();
-				}
-				$query = NULL;
-			break;
-
-			case Jam_Association::ERASE:
-				$query = Jam_Query_Builder_Delete::factory($this->foreign_model)
-					->where(':primary_key', 'IN', $ids);
-
-				if ($this->is_polymorphic())
-				{
-					$query->value($this->polymorphic_key, NULL);
-				}
-			break;
+			foreach (Jam::all($this->foreign_model)->where_key($ids) as $item )
+			{
+				$item->delete();
+			}
+			$query = NULL;
+		}
+		elseif ($this->delete_on_remove === Jam_Association::ERASE)
+		{
+			$query = Jam_Query_Builder_Delete::factory($this->foreign_model)
+				->where(':primary_key', 'IN', $ids);
```

Comment:
@hkdobrev This is checked beforehand wherever this method is used, thats why didn't put a check in it.